MIP/mip-16/README.md

```diff
@@ -0,0 +1,108 @@
+
+In the sequel we assume:
+- \$mMOVE is the wrapped \$MOVE token on Movement Rollup.
+- \$MOVE is used to pay the gas fees on Movement Rollup. 
+- one \$mMOVE has the same value as one \$MOVE.
+- some data are published to a _data availability (DA)_ layer and some to Ethereum mainnet (L1). 
+- we execute MoveVM transactions only (not EVM transactions).
+
+The cost of processing a transaction on Movement Rollup is [multi-dimensional](https://medium.com/offchainlabs/understanding-arbitrum-2-dimensional-fees-fd1d582596c9):
+
+1. **execution fees**: off-chain/Movement, an amount that is expressed in the Movement native tokens,
+    \$MOVE, and
+2. **data availability** fees: correspond to posting data to the availability layer, expressed in \$ETH.
+3. **L1** fees: correspond to transactions on Ethereum mainnet, expressed in \$ETH.
+
+### Execution fees on movement
+
+Movement Rollup is a Move-based rollup. As a result a transaction $\textit{tx}$ executed on Movement Rollup consumes a certain amount of gas, 
+$g(tx)$. We want to express this fees in \$MOVE and to do so we need to define the price of a unit of gas on Movement Rollup. This price may evolve with time or per block production, and we let $gpMvmt(k)$ be the price of a unit of gas at block $k$.
+
+If a transaction $tx$ is included in block $k$ and its execution consumes $g(tx)$ gas units, the corresponding _execution_ fees (for the transaction issuer) are:
+
+$$ g(tx) \times gpMvmt(k)\mathpunct.$$
+
+### Data availability costs
+
+On a rollup, transactions are executed in batches. The transactions data and the states have to be made available so that third parties can audit them. This is achieved by publishing the data (transaction data and states) to a data availability layer.
+
+The cost of publication may depend on the size of the data and we assume that this is fixed per bytes (it may change from time to time but is fixed for a rather long period, e.g. 1 month).
+
+**What do we have to publish?** Some information that enables third parties to reproduce a the execution of a transaction and check whether it is valid.
+This means we have to publish the _calldata_ (transaction and its parameters) and some information about the _source_ (previous) and _target_ (new) states. It could be the new state or the change set that completely characterises the new state given the previous state. 
+
+The DA costs of a transaction consist of two components:
+
+- transaction data, a fixed cost that depends on the size of the calldata $|tx|$ of the transaction $tx$.
+
+- a fraction of the cost of the size of new state (or change set). Indeed, transcations are processed in batches in a block, so a new block/state corresponds to the exexcution of a sequence of transactions. If a block $B_k$ (resp. change set $S_k$) has size $|B_k|$ (resp. $S_k$) and contains $n_k$ transactions, each transaction in the block incurs a fraction $|B_k|/n_k$ (resp. $|S_k|/n_k$) of the DA cost for $B_k$ (resp. $S_k$). 
```

Comment:
I would say this is better regarded as an execution cost. 

The DA just contains transaction data; change sets are not recorded here. Regardless of underlying implementation, this should remain so until the DA itself has a programmatic layer--e.g., contingency plan execution in an atomic transaction bundle. 

Similarly, because the size of the state root is constant, the size of the change set does not need to be modeled for settlement.

Reply:
So where do we publish the actual state or change set?
Do we only publish state roots? How can a user verify a state transition if we only publish the state roots? 
They would need to replay all the Txs from the genesis state.

Reply:
> So where do we publish the actual state or change set?

The only thing we publish currently is the state root which digests information about the change set but not purely. 

> How can a user verify a state transition if we only publish the state roots?

A user cannot verify a transition in the same sense that they would in a ZK or Optimistic (fraud-proof-based) rollup. 

If they wanted to verify a transition and avoid replaying from genesis, they would (1) pick an accepted state root for their given fork choice, (2) sync and verify the appropriate state at that point, and then (3) replay from there until the transition they will attempt to verify. 

Because votes on a given commitment are final in FFM, the user can only use this verification to decide their participation in a fork. They cannot prevent a faulty commitment--as in the ZK case--nor revert a faulty commitment--as in the Optimistic case. 

A request to model decision-making w.r.t. both temporary and long-lived forks has been submitted in [MD-3.D3](https://github.com/movementlabsxyz/MIP/tree/l-monninger/mcr-asynchrony/MD/md-3#d3-address-asynchronous-upgrades-fork-creation-and-fork-stake-problems).